### PR TITLE
feat: add Electra support to light client functions

### PR DESCRIPTION
**Pull request by rupam-04**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Bug fix
> Feature
> Documentation
> Other

**What does this PR do? Why is it needed?**

Adds Electra to switch case in light client functions. Anything new not required for Electra as specs have mostly stayed the same since Deneb

**Which issues(s) does this PR fix?**

Needed for #12991 

**Other notes for review**

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

File: beacon-chain/core/light-client/lightclient.go [hunk 1]
```diff
@@ -423,7 +423,7 @@ func blockToLightClientHeaderCapella(ctx context.Context, block interfaces.ReadO
 
 func blockToLightClientHeaderDeneb(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeaderDeneb, error) {
 	if block.Version() != version.Deneb {
```
Comment (rkapka) [inline beacon-chain/core/light-client/lightclient.go:425]:
```suggestion
	if block.Version() < version.Deneb {
```

**Review by rkapka — CHANGES_REQUESTED** [review]

**Comment by rkapka** [comment]
Please add Electra test cases

File: beacon-chain/rpc/eth/light-client/helpers.go [hunk 2]
```diff
@@ -226,8 +226,13 @@ func createLightClientBootstrapDeneb(ctx context.Context, state state.BeaconStat
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get current sync committee proof")
 	}
-
-	branch := make([]string, fieldparams.NextSyncCommitteeBranchDepth)
+	var branch []string
+	switch block.Version() {
```
Comment (rupam-04) [inline beacon-chain/rpc/eth/light-client/helpers.go:230]: for Electra the value of `range currentSyncCommitteeProof` is coming out to be 6 whereas in Deneb it is 5. I am not sure if this is a good way to define the size. Open to suggestions

Comment (rkapka) [reply beacon-chain/rpc/eth/light-client/helpers.go:230]: We should have a new constant for that - `NextSyncCommitteeBranchDepthElectra`

File: config/fieldparams/mainnet.go [hunk 3]
```diff
@@ -48,4 +48,5 @@ const (
 	MaxDeposits                           = 16            // Maximum number of deposits in a block.
 	MaxVoluntaryExits                     = 16            // Maximum number of voluntary exits in a block.
 	MaxBlsToExecutionChanges              = 16            // Maximum number of bls to execution changes in a block.
+	NextSyncCommitteeBranchDepthElectra   = 6             // NextSyncCommitteeBranchDepthElectra defines the depth of the next sync committee branch for Electra.
```
Comment (rkapka) [inline config/fieldparams/mainnet.go:51]: - please add it next to the the pre-Electra one
- you need to add it to `minimal.go` too

**Review by rkapka — CHANGES_REQUESTED** [review]

File: config/fieldparams/mainnet.go [hunk 4]
```diff
@@ -34,6 +34,7 @@ const (
 	BlobSidecarSize                       = 131928        // defined to match blob sidecar size in bazel ssz codegen
 	KzgCommitmentInclusionProofDepth      = 17            // Merkle proof depth for blob_kzg_commitments list item
 	NextSyncCommitteeBranchDepth          = 5             // NextSyncCommitteeBranchDepth defines the depth of the next sync committee branch.
+	NextSyncCommitteeBranchDepthElectra   = 6             // NextSyncCommitteeBranchDepthElectra defines the depth of the next sync committee branch.
```
Comment (rupam-04) [inline config/fieldparams/mainnet.go:37]: removed the "for Electra" from the comment as other Electra constants don't have it either

**Review by rkapka — APPROVED** [review]

**Pull request merged**